src/otx/v2/adapters/torch/mmengine/mmseg/engine.py

```diff
@@ -0,0 +1,168 @@
+"""OTX adapters.torch.mmengine.mmpretrain.Engine API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import torch
+from mmseg.apis import MMSegInferencer
+from mmseg.registry import VISUALIZERS
+
+from otx.v2.adapters.torch.mmengine.engine import MMXEngine
+from otx.v2.adapters.torch.mmengine.mmseg.registry import MMSegmentationRegistry
+from otx.v2.adapters.torch.mmengine.modules.utils.config_utils import CustomConfig as Config
+from otx.v2.api.utils.logger import get_logger
+
+if TYPE_CHECKING:
+    import numpy as np
+
+logger = get_logger()
+
+
+class MMSegEngine(MMXEngine):
+    """The MMPretrainEngine class is responsible for running inference on pre-trained models."""
+
+    def __init__(
+        self,
+        work_dir: str | Path | None = None,
+    ) -> None:
+        """Initialize a new instance of the MMPretrainEngine class.
+
+        Args:
+            work_dir (Optional[Union[str, Path]], optional): The working directory for the engine. Defaults to None.
+            config (Optional[Union[Dict, Config, str]], optional): The configuration for the engine. Defaults to None.
+        """
+        super().__init__(work_dir=work_dir)
+        self.registry = MMSegmentationRegistry()
+        self.visualizer = {"name": "visualizer", "type": "SegLocalVisualizer"}
+        self.evaluator = {"type": "IoUMetric", "iou_metrics": ["mDice"]}
+
+    def _update_config(self, func_args: dict, **kwargs) -> tuple[Config, bool]:
+        """Update the configuration of the runner with the provided arguments.
+
+        Args:
+            func_args (dict): The arguments passed to the engine.
+            **kwargs: Additional keyword arguments to update the configuration for mmengine.Runner.
+
+        Returns:
+            tuple[Config, bool]: Config, True if the configuration was updated, False otherwise.
+        """
+        config, update_check = super()._update_config(func_args, **kwargs)
+        if getattr(config, "val_dataloader", None) and not hasattr(config.val_evaluator, "type"):
+            config.val_evaluator = self.evaluator
+            config.val_cfg = {"type": "ValLoop"}
+        if getattr(config, "test_dataloader", None) and not hasattr(config.test_evaluator, "type"):
+            config.test_evaluator = self.evaluator
+            config.test_cfg = {"type": "TestLoop"}
+        if hasattr(config, "visualizer") and config.visualizer.type not in VISUALIZERS:
+            config.visualizer = self.visualizer
```

Comment:
I have a similar comment on PR for mmdet in v2 https://github.com/openvinotoolkit/training_extensions/pull/2598/files#r1387937126. Isn't is possible to do this with adequate `func_args`?

Reply:
your suggestion sounded good, but when I tried it out, it didn't quite do the trick. I don't see much point in sticking with the current `_update_config` design — it's kind of messy, and updating parameters isn't clear.

 Instead of tweaking the config here, why not just load a YAML file that spells out everything we need? I think @sungmanc already proposed refactoring this part.

Reply:
sounds great. let's merge as it is for now first and try to redesign configurations overall later.